hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1303,8 +1303,6 @@ def test_not_equal(x, y):
         """
         st = self.shrink_target
         pieces = [st.buffer[c.start : c.end] for c in ex.children]
-        if not pieces:
-            pieces = [st.buffer[ex.start : ex.end]]
```

Comment:
Is this now unused because the step-lambda already rules out examples with no children?

Reply:
Exactly, yes.

Reply:
Although also it was useless in this instance.